### PR TITLE
transfer leads initial version

### PR DESCRIPTION
**Pull request by congocongo**
@philfreo  for #40

### CODE REVIEW

File: scripts/transfer_leads.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+""")
+parser.add_argument('--api-key', '-k', required=True, help='API Key')
+parser.add_argument('--target-api-key', required=True, help='Target company API Key')
+parser.add_argument('--development', '-d', action='store_true',
+                    help='Use a development (testing) server rather than production.')
+parser.add_argument('--delete', action='store_true',
+                    help='Delete transfered leads.')
+parser.add_argument('--confirmed', '-c', action='store_true',
+                    help='Without this flag, the script will do a dry run without actually updating any data.')
+parser.add_argument('query', help='Search query')
```
Comment (philfreo) [inline scripts/transfer_leads.py]: this script won't work properly unless there's a `sort:date`. either put this in the help message or ensure it in code

File: scripts/transfer_leads.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+    # tasks
+    new_tasks = []
+    for task in lead['tasks']:
+        new_tasks.append({
+            'lead_id': new_lead['id'],
+            'text': task['text'],
+            'due_date': task['due_date'],
+            'is_complete': task['is_complete']
+        })
```
Comment (philfreo) [inline scripts/transfer_leads.py]: you need to copy over the `assigned_to` field (which is which user the task is assigned to)

File: scripts/transfer_leads.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+                    'attachments': activity['attachments'],
+                    'subject': activity['subject'],
+                    'body_text': activity['body_text'],
+                    'body_html': activity['body_html'],
+                    'sender': activity['sender'],
+                    'to': activity['to'],
+                    'cc': activity['cc'],
+                    'bcc': activity['bcc'],
+                    'envelope': activity['envelope'],
+                    'need_smtp_credentials': activity['need_smtp_credentials'],
```
Comment (philfreo) [inline scripts/transfer_leads.py]: don't include `need_smtp_credentials` or `send_attempts`

File: scripts/transfer_leads.py [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+        has_more = resp['has_more']
+
+    for activity in new_lead_activities:
+        if activity['_type'] == 'Note':
+            if args.confirmed:
+                target_api.post('activity/note', data={
+                    'lead_id': new_lead['id'],
+                    'note': activity['note'],
+                    'date_created': activity['date_created'],
+                    'date_updated': activity['date_updated']
```
Comment (philfreo) [inline scripts/transfer_leads.py]: you should include user_id

File: scripts/transfer_leads.py (hunk 3 above)
Comment (philfreo) [inline scripts/transfer_leads.py]: include user_id

File: scripts/transfer_leads.py [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+                    'status': activity['status'],
+                    'remote_phone': activity['remote_phone'],
+                    'phone': activity['phone'],
+                    'local_phone': activity['local_phone'],
+                    'transferred_from': activity['transferred_from'],
+                    'transferred_to': activity['transferred_to'],
+                    'recording_url': activity['recording_url'],
+                    'date_created': activity['date_created'],
+                    'date_updated': activity['date_updated']
+                })
```
Comment (philfreo) [inline scripts/transfer_leads.py]: include user_id if it exists in the new org

File: scripts/transfer_leads.py [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+                    'updated_by_name': activity['updated_by_name'],
+                    'voicemail_duration': activity['voicemail_duration'],
+                    'note': activity['note'],
+                    'source': activity['source'],
+                    'status': activity['status'],
+                    'remote_phone': activity['remote_phone'],
+                    'phone': activity['phone'],
+                    'local_phone': activity['local_phone'],
+                    'transferred_from': activity['transferred_from'],
+                    'transferred_to': activity['transferred_to'],
```
Comment (philfreo) [inline scripts/transfer_leads.py]: only include transferred_from/to if those user_ids exist in new org

File: scripts/transfer_leads.py [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+                })
+            logging.info('target: %s added email: %s %s' % (new_lead['id'], activity['to'], activity['subject']))
+        elif activity['_type'] == 'Call':
+            if args.confirmed:
+                target_api.post('activity/call', data={
+                    'lead_id': new_lead['id'],
+                    'created_by_name': activity['created_by_name'],
+                    'direction': activity['direction'],
+                    'duration': activity['duration'],
+                    'updated_by_name': activity['updated_by_name'],
```
Comment (philfreo) [inline scripts/transfer_leads.py]: this is a read-only field, don't include it

File: scripts/transfer_leads.py [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+                    'send_attempts': activity['send_attempts'],
+                    'date_created': activity['date_created'],
+                    'date_updated': activity['date_updated']
+                })
+            logging.info('target: %s added email: %s %s' % (new_lead['id'], activity['to'], activity['subject']))
+        elif activity['_type'] == 'Call':
+            if args.confirmed:
+                target_api.post('activity/call', data={
+                    'lead_id': new_lead['id'],
+                    'created_by_name': activity['created_by_name'],
```
Comment (philfreo) [inline scripts/transfer_leads.py]: readonly, don't include it

File: scripts/transfer_leads.py [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+                    'date_created': activity['date_created'],
+                    'date_updated': activity['date_updated']
+                })
+            logging.info('target: %s added note: %s' % (new_lead['id'], activity['note']))
+
+        elif activity['_type'] == 'Email':
+            if args.confirmed:
+                target_api.post('activity/email', data={
+                    'lead_id': new_lead['id'],
+                    'status': 'sent',
```
Comment (philfreo) [inline scripts/transfer_leads.py]: some emails have other statuses (e.g. incoming emails have status `inbox` I think)

**Comment by congocongo** [comment]
@philfreo updated

File: scripts/transfer_leads.py (hunk 1 above)
Comment (congocongo) [inline scripts/transfer_leads.py]: I added something like:

```python
if not 'sort:created' in args.query:
    lead_query = args.query + ' sort:created'
else:
    lead_query = args.query

File: scripts/transfer_leads.py (hunk 6 above)
Comment (philfreo) [inline scripts/transfer_leads.py]: @congocongo what about this?

**Pull request closed without merging**